### PR TITLE
feat: talos support

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/3161

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,10 @@ module github.com/longhorn/longhorn-instance-manager
 
 go 1.20
 
+replace github.com/longhorn/longhorn-engine => github.com/c3y1huang/longhorn-engine v1.4.0-rc1.0.20230818103126-3ba57dbd5d39
+
+replace github.com/longhorn/go-common-libs => github.com/c3y1huang/go-common-libs v0.0.0-20230818102433-680b5db81371
```
Comment (c3y1huang) [inline go.mod:7]: TODO: update after https://github.com/longhorn/go-common-libs/pull/1

File: go.mod [hunk 2]
```diff
@@ -2,6 +2,10 @@ module github.com/longhorn/longhorn-instance-manager
 
 go 1.20
 
+replace github.com/longhorn/longhorn-engine => github.com/c3y1huang/longhorn-engine v1.4.0-rc1.0.20230818103126-3ba57dbd5d39
```
Comment (c3y1huang) [inline go.mod:5]: TODO: update after https://github.com/longhorn/longhorn-engine/pull/919

**Review by innobead — APPROVED** [review]
Just wait for longhorn-engine PR merge first.

**Review by innobead — CHANGES_REQUESTED** [review]
Please update the engine dep.

**Review by innobead — APPROVED** [review]

**Pull request merged**